R/explore.R

```diff
@@ -0,0 +1,33 @@
+#'  Launches shiny application based on data from [tune::fit_resamples()]
+#'
+#' This function takes the data based on [tune::fit_resamples()] results to
+#' launch a Shiny app. This function normally does not return; interrupt R
+#' to stop the app (usually by pressing Ctrl+C or Esc).
+#' @param x The [tune::fit_resamples()] result.
+#' @param hover_cols The columns to display while hovering in the Shiny app.
+#' @param hover_only A Boolean to turn on and off hovering in the interactive
+#' plots.
+#' @param ... Other parameters not currently used.
+#' @keywords models, regression, graphs, classes, classif
+#' @export
+
+explore <- function(x, hover_cols = NULL, hover_only = NULL, ...) {
+  UseMethod("explore")
+}
+
+#' @export
+#' @rdname explore
+explore.default <- function(x, hover_cols = NULL, hover_only = NULL, ...) {
+  rlang::abort("No `explore()` exists for this type of object.")
+}
+
+#' @export
+#' @rdname explore
+explore.tune_results <-
+  function(x,
+           hover_cols = NULL,
+           hover_only = NULL,
+           ...) {
+    dat <- organize_data(x, hover_cols, ...)
```

Comment:
Since `organize_data()` uses `enquo()` on `hover_cols`, I think you just need to use the `!!enquo()` pattern in `explore.tune_results()` to pass it along as something that `organize_data()` can capture correctly.

So you can either do:

```r
explore.tune_results <-
  function(x,
           hover_cols = NULL,
           hover_only = NULL,
           ...) {
    hover_quo <- enquo(hover_cols)
    dat <- organize_data(x, hover_cols = !!hover_quo, ...)
    shiny_models(dat, hover_only)
  }
```

Or you can use the relatively new-ish `{{ x }}` ("curly curly") pattern, which is identical to calling `!!enquo(x)`. I would recommend this, as this is what `{{ }}` was created for:

```r
explore.tune_results <-
  function(x,
           hover_cols = NULL,
           hover_only = NULL,
           ...) {
    dat <- organize_data(x, hover_cols = {{ hover_cols }}, ...)
    shiny_models(dat, hover_only)
  }
```

Reply:
I don't think this was meant to be marked as resolved yet 😬